dags/outcome_dag.py

```diff
@@ -0,0 +1,138 @@
+    def branch_term(**kwargs):
+        # Sourced from: https://stackoverflow.com/questions/58976215/airflow-scheduling-how-to-run-initial-setup-task-only-once
+        date = kwargs['data_interval_start']
+        if date == dag.end_date:
+            return 'delete_cluster'
+        else:
+            return 'skip_task_term'
+    branch_task_init = BranchPythonOperator(
+        task_id='branch_task_init', 
+        python_callable=branch_init,
+        provide_context=True
+    )
+    create_cluster = DataprocCreateClusterOperator(
+        task_id='create_cluster',
+        project_id=os.environ['GCP_PROJECT_ID'],
+        cluster_config=CLUSTER_CONFIG,
+        region='us-central1',
+        cluster_name=os.environ['GCP_CLUSTER_NAME']
+    )
+    skip_task_init = DummyOperator(
+        task_id='skip_task_init'
+    )
+    extract = PythonOperator(
+        task_id="extract",
+        python_callable=extract,
+        trigger_rule='one_success',
+        op_kwargs={
+            'target_file': EXTRACT_TARGET_FILE,
+            'start_date': '2023-11-01',
+            'execution_date': '{{ ds }}'
+        }
+    )
+    transform = DataprocSubmitJobOperator(
+        task_id="transform", 
+        job=PYSPARK_JOB, 
+        region='us-central1', 
+        project_id=os.environ['GCP_PROJECT_ID']
+    )
+    load_outcome_animal_dim = PythonOperator(
+        task_id="load_outcome_animal_dim",
+        python_callable=load_data,
+        op_kwargs={
+            'table_file': TRANSFORM_TARGET_DIR + '/outcome_animal_dim.parquet',
+            'table_name': 'outcome_animal_dim',
+            'key': 'animal_id'
+        }
+    )
+    load_outcome_date_dim = PythonOperator(
+        task_id="load_outcome_date_dim",
+        python_callable=load_data,
+        op_kwargs={
+            'table_file': TRANSFORM_TARGET_DIR + '/outcome_date_dim.parquet',
+            'table_name': 'outcome_date_dim',
+            'key': 'outcome_date_id'
+        }
+    )
+    load_outcome_type_dim = PythonOperator(
+        task_id="load_outcome_type_dim",
+        python_callable=load_data,
+        op_kwargs={
+            'table_file': TRANSFORM_TARGET_DIR + '/outcome_type_dim.parquet',
+            'table_name': 'outcome_type_dim',
+            'key': 'outcome_type_id'
+        }
+    )
+    load_outcome_fact_table = PythonOperator(
+        task_id="load_outcome_fact_table",
+        python_callable=load_fact_table,
+        op_kwargs={
+            'table_file': TRANSFORM_TARGET_DIR + '/outcome_fact_table.parquet',
+            'table_name': 'outcome_fact_table'
+        }
+    )
+    branch_task_term = BranchPythonOperator(
+        task_id='branch_task_term', 
+        python_callable=branch_term,
+        provide_context=True
+    )
+    delete_cluster = DataprocDeleteClusterOperator(
+        task_id="delete_cluster",
+        project_id=os.environ['GCP_PROJECT_ID'],
+        cluster_name=os.environ['GCP_CLUSTER_NAME'],
+        region='us-central1',
```

Comment:
right now, if something goes wrong between creating and deleting the cluster won't be deleted. To avoid this, you can use trigger_rule="all_done"